builtin/providers/dockercloud/resource_dockercloud_node_cluster.go

```diff
@@ -0,0 +1,255 @@
+package dockercloud
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/go-dockercloud/dockercloud"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceDockercloudNodeCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDockercloudNodeClusterCreate,
+		Read:   resourceDockercloudNodeClusterRead,
+		Update: resourceDockercloudNodeClusterUpdate,
+		Delete: resourceDockercloudNodeClusterDelete,
+		Exists: resourceDockercloudNodeClusterExists,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"node_provider": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"size": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"disk": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"node_count": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceDockercloudNodeClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	provider := d.Get("node_provider").(string)
+	region := d.Get("region").(string)
+	size := d.Get("size").(string)
+
+	opts := dockercloud.NodeCreateRequest{
+		Name:     d.Get("name").(string),
+		Region:   fmt.Sprintf("/api/infra/v1/region/%s/%s/", provider, region),
+		NodeType: fmt.Sprintf("/api/infra/v1/nodetype/%s/%s/", provider, size),
+	}
+
+	if attr, ok := d.GetOk("disk"); ok {
+		opts.Disk = attr.(int)
+	}
+
+	if attr, ok := d.GetOk("node_count"); ok {
+		opts.Target_num_nodes = attr.(int)
+	}
+
+	tags := d.Get("tags.#").(int)
```

Comment:
It was getting the length of the tag list. Will change to a more straightforward loop over `[]interface{}`.